nncf/common/pruning/operations.py

```diff
@@ -320,20 +318,47 @@ def _is_flatten(node: NNCFNode) -> bool:
         return len(node.layer_attributes.output_shape) == 2
 
     @staticmethod
-    def _is_not_mixing_dim(node: NNCFNode, graph) -> bool:
-        input_shape = node.layer_attributes.input_shape
+    def _is_not_mixing_dim(node: NNCFNode) -> bool:
+        def _check_dim_splitted(dim_from: int, dims_to: List[int],
+                                dims_to_start_idx: int):
```

Comment:
This function is tying to project `dim_from` to several elements into `dim_to` starting from `dims_to_start_idx`